src/routes/ConsortiumManager/settings/authorization-roles/UsersCapabilities/UsersCapabilitiesCompare/UsersCapabilitiesCompare.js

```diff
@@ -0,0 +1,103 @@
+import {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  FormattedMessage,
+  useIntl,
+} from 'react-intl';
+import {
+  useLocation,
+  useHistory,
+} from 'react-router-dom';
+
+import {
+  Layer,
+  Pane,
+  PaneHeader,
+  Paneset,
+} from '@folio/stripes/components';
+
+import { AUTHORIZATION_ROLES_ROUTE } from '../../../../../../constants';
+import { useConsortiumManagerContext } from '../../../../../../contexts';
+import { TENANT_ID_SEARCH_PARAMS } from '../../../users/general/PermissionSets/constants';
+import { COMPARE_ITEM_NAME } from '../../../users/general/PermissionSets/PermissionSetsCompare/constants';
+import { UsersCapabilitiesCompareItems } from '../UsersCapabilitiesCompareItems';
+
+export const UsersCapabilitiesCompare = () => {
+  const history = useHistory();
+  const { search } = useLocation();
+  const intl = useIntl();
+
+  const initialSelectedMemberId = useMemo(() => new URLSearchParams(search).get(TENANT_ID_SEARCH_PARAMS), [search]);
+  const [rolesToCompare, setRolesToCompare] = useState({
+    [COMPARE_ITEM_NAME.LEFT_COLUMN]: [],
+    [COMPARE_ITEM_NAME.RIGHT_COLUMN]: [],
+  });
+
+  const { selectedMembers } = useConsortiumManagerContext();
+
+  const handleRolesToCompare = useCallback((roles, columnName) => {
+    setRolesToCompare({
```

Comment:
If you want to set a new state based on the previous one, please use an update function instead then and remove `rolesToCompare` hook dep:
```
setRolesToCompare((prevRoles) => ({
  ...prevRoles,
  ..
}))
```

Reply:
Can you explain why we can't do it through callback is it done here https://github.com/folio-org/ui-consortia-settings/blob/master/src/routes/ConsortiumManager/settings/users/general/PermissionSets/PermissionSetsCompare/PermissionSetsCompare.js#L39 ? The same logic

Reply:
Agree, at the aforementioned place, it wouldn't hurt to make a correction, as this does not comply with the recommendations of React team. The code mentioned above is just a temporary copy-paste from ui-users and we expect to remove that code and use shared users components in the future.

Reply:
the reason why we shouldn't update a state that way is described here https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state